src/utils/json/flatten.rs

```diff
@@ -273,52 +274,83 @@ pub fn flatten_array_objects(
 /// Recursively flattens a JSON value.
 /// - If the value is an array, it flattens all elements of the array.
 /// - If the value is an object, it flattens all nested objects and arrays.
+/// - If the JSON value is heavily nested (with more than 4 levels of hierarchy), returns error
 /// - Otherwise, it returns the value itself in a vector.
 ///
 /// Examples:
 /// 1. `{"a": 1}` ~> `[{"a": 1}]`
 /// 2. `[{"a": 1}, {"b": 2}]` ~> `[{"a": 1}, {"b": 2}]`
 /// 3. `[{"a": [{"b": 1}, {"c": 2}]}]` ~> `[{"a": {"b": 1)}}, {"a": {"c": 2)}}]`
-/// 3. `{"a": [{"b": 1}, {"c": 2}], "d": {"e": 4}}` ~> `[{"a": {"b":1}, "d": {"e":4}}, {"a": {"c":2}, "d": {"e":4}}]`
-pub fn flatten_json(value: &Value) -> Vec<Value> {
+/// 4. `{"a": [{"b": 1}, {"c": 2}], "d": {"e": 4}}` ~> `[{"a": {"b":1}, "d": {"e":4}}, {"a": {"c":2}, "d": {"e":4}}]`
+/// 5. `{"a":{"b":{"c":{"d":{"e":["a","b"]}}}}}` ~> returns error - heavily nested, cannot flatten this JSON
+pub fn flatten_json(value: &Value) -> Result<Vec<Value>, anyhow::Error> {
+    if has_more_than_four_levels(value, 1) {
+        return Err(anyhow!("heavily nested, cannot flatten this JSON"));
```

Comment:
Why error out? Why not just not flatten and keep the type as is?

Reply:
i.e. `{"a": [{"b":{"c":{"d":["e","f"]}}}, "c": {}]}` ~> `[{"a":{"b":{"c":{"d":["e","f"]}}}}, {"a":{"c": {}}}]`